### DISCUSSION TITLE
Is it ok to use nested q-layout?

### DISCUSSION THREAD

**Discussion by Shirockov55** [quasarframework/quasar#15831] [General - Components / Directives / etc] [answered]
Is it ok to use nested q-layout? For example, there is a page design that largely depends on the height of the browser window. This works well with root elements. But what if inside the page we need viewports that also need to calculate the residual height of the region, like the q-page component with style-fn property does? Could this be breaking something or is nested q-layout components normal supported behavior?

**Comment by suleimanas** [comment, marked as answer]
you don't need it but yes you can have containerized Qlayout inside the root Qlayout
you can use **container** prop
https://quasar.dev/layout/layout#containerized-qlayout
